File: src/main/java/com/brightcove/johnny/CharOracle.java
package com.brightcove.johnny;

/**
 * Represents a set of characters, either positive or negative.
 * (Does not provide a way to characterize or enumerate the set.)
 */
public interface CharOracle {
    /** True iff set contains the character. */
    boolean containsChar(char c);

    /** True iff set contains the code point. */
    boolean containsCodePoint(int cp);
}

File: src/main/java/org/timmc/johnny/internal/CharOracle.kt
package org.timmc.johnny.internal;

/**
 * Represents a set of characters, either positive or negative.
 * (Does not provide a way to characterize or enumerate the set.)
 */
interface CharOracle {
    /** True iff set contains the character. */
    fun containsChar(c: Char): Boolean

    /** True iff set contains the code point. */
    fun containsCodePoint(cp: Int): Boolean
}
